src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -46,11 +45,9 @@ public User add(User user) {
     @Override
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
-            Query<User> userQuery = session
-                    .createQuery("from User where email =:email",
-                    User.class);
-            userQuery.setParameter("email", email);
-            return userQuery.uniqueResultOptional();
+            return session.createQuery("SELECT u FROM User u"
+                    + " LEFT JOIN FETCH u.roles WHERE u.email = :email", User.class)
```

Comment:
inner join would better here, there should not be situation when user doesn't have role

Reply:
you didnt fix this comment - change LEFT JOIN to JOIN and don't remove .setParameter()